.github/workflows/ASSIGN_MILESTONE.yml

```diff
@@ -0,0 +1,55 @@
+name: Assign Milestones
+
+on:
+  issues:
+    types: ['closed']
+  pull_request:
+    types: ['closed']
+
+jobs:
+  assignMilestone:
+    runs-on: ubuntu-latest
+    name: Assign Milestone
+    steps:
+      - name: Get current Milestone
+        id: getMilestone
+        env: 
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          # Fetch the list of milestones for the repository
+          # Filter for milestones that start with 'M' and are open
+          MILESTONE=$(gh api -H "Accept: application/vnd.github.v3+json" \
+            /repos/${{ github.repository }}/milestones \
+            --jq '.[] | select(.title | startswith("M")) | .number'
+          )
+          
+          echo "MILESTONE_NUMBER=$MILESTONE" >> $GITHUB_OUTPUT
+      - name: Assign Issue
+        if: |
+          github.event.issue &&
+          github.event.issue.state_reason != 'not_planned' &&
```

Comment:
Shall we check who closed the issue? I'd like to avoid a situation when a 3rd party can assign milestones.

Reply:
[9ccf5ea](https://github.com/camunda/camunda-modeler/pull/4400/commits/9ccf5ea792c86e153cf0aee78ce7625b8226f6cb) adds a check to ensure we only assign milestones when the user has at least write permissions to the repo

Reply:
Great idea with the permissions check :)